test/unit/index.js

```diff
@@ -48,6 +41,55 @@ describe("asyncPool", function() {
         await asyncPool(2, [100, 500, 300, 200], timeout);
         expect(results).to.deep.equal([100, 300, 500, 200]);
       });
+
+      it("runs all promises in parallel when the pool is bigger than needed", async function() {
+        const results = [];
+        const timeout = i =>
+          new Promise(resolve =>
+            setTimeout(() => {
+              results.push(i);
+              resolve();
+            }, i)
+          );
+        await asyncPool(5, [100, 500, 300, 200], timeout);
+        expect(results).to.deep.equal([100, 200, 300, 500]);
+      });
+
+      it("rejects on error with pool limit effective", async function() {
+        const results = [];
```

Comment:
Do we use results?

Reply:
No, this was a copy&paste left-over. Will remove it.

Reply:
:+1: thanks